src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -63,9 +72,23 @@ public List<BookDto> search(BookSearchParametersDto searchParameters) {
                 .toList();
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findBooksByCategory(Pageable pageable, Long id) {
+        return bookRepository.findAllByCategoryId(id).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
     private void checkIfBookExistsById(Long id) {
         if (!bookRepository.existsById(id)) {
             throw new EntityNotFoundException("Can't find book with id " + id);
         }
     }
+
+    private void setCategories(CreateBookRequestDto requestDto, Book book) {
+        Set<Category> categorySet = requestDto.categoryIds().stream()
+                .map(categoryRepository::getReferenceById)
+                .collect(Collectors.toSet());
```

Comment:
does it work? get reference returns a proxy with just id

when you save it to set it invokes hashCode, which in turn has to cause LazyInitException

does it really work for you?

Reply:
worked, had open-in-view = true)